tests/run-macros/beta-reduce-inline-result/Macro_1.scala

```diff
@@ -4,13 +4,15 @@ object Macros {
   inline def betaReduce[Arg,Result](inline fn : Arg=>Result)(inline arg: Arg): Result =
     ${ betaReduceImpl('{ fn })('{ arg }) }
 
-  def betaReduceImpl[Arg,Result](fn: Expr[Arg=>Result])(arg: Expr[Arg])(using qctx : QuoteContext): Expr[Result] =
-    Expr.betaReduce(fn)(arg)
+  def betaReduceImpl[Arg: Type, Result: Type](fn: Expr[Arg=>Result])(arg: Expr[Arg])(using qctx : QuoteContext): Expr[Result] =
+    val app = '{$fn($arg)}
+    Expr.betaReduce(app).getOrElse(app)
```

Comment:
If the most common usecase is to do betaReduce(x).getOrElse(x), wouldn't it be nicer to make betaReduce return its input in case it couldn't do any beta-reduction? It would still be possible to explicitly check if it did something by comparing the input and output using `eq`

Reply:
Done